api/oci/extensions/repositories/artifactset/artifactset.go

```diff
@@ -204,26 +206,123 @@ func (a *namespaceContainer) AddTags(digest digest.Digest, tags ...string) error
 	defer a.base.Unlock()
 
 	idx := a.GetIndex()
-	for i, e := range idx.Manifests {
-		if e.Digest == digest {
-			if e.Annotations == nil {
-				e.Annotations = map[string]string{}
-				idx.Manifests[i].Annotations = e.Annotations
-			}
-			cur := RetrieveTags(e.Annotations)
-			if cur != "" {
-				cur = strings.Join(append([]string{cur}, tags...), ",")
-			} else {
-				cur = strings.Join(tags, ",")
-			}
-			e.Annotations[TAGS_ANNOTATION] = cur
-			if a.base.FileSystemBlobAccess.Access().GetInfo().GetDescriptorFileName() == OCIArtifactSetDescriptorFileName {
-				e.Annotations[OCITAG_ANNOTATION] = tags[0]
+	isOCI := a.base.FileSystemBlobAccess.
+		Access().
+		GetInfo().
+		GetDescriptorFileName() == OCIArtifactSetDescriptorFileName
```

Comment:
Just structure-wise - this variable could be set right before it is used on L256